client_handler.cpp

```diff
@@ -1854,35 +1857,38 @@ bool ClientHandler::OnRequestMediaAccessPermission(
 		return accepted;
 	}
 
-	CString confirmMessage;
-	CString enableMediaConfirmation;
+	HWND hWindow = GetSafeParentWnd(browser);
 	CString requestOrigin = (LPCWSTR)requesting_origin.c_str();
-	enableMediaConfirmation.LoadString(ID_ENABLE_MEDIA_CONFIRMATION);
-	confirmMessage.Format(enableMediaConfirmation, (LPCTSTR)requestOrigin);
-	confirmMessage += "\n";
+	if (theApp.m_AppSettings.GetMediaAccessPermission() == AppSettings::MediaAccessPermission::MANUAL_MEDIA_APPROVAL)

```

Comment:
We don't need to check and add the cache when `MediaAccessPermission` is not MANUAL_MEDIA_APPROVAL.

So how about moving these lines inside of this if block?

https://github.com/ThinBridge/Chronos/pull/134/files#diff-8aad731b958e24ed51cb70eac250f7763e1f8110fadf4608f54c205a01a6845bR1848-R1858
```
	std::tuple<CefString, uint32> permissionInfo = std::tie(requesting_origin, requested_permissions);
	std::map<std::tuple<CefString, uint32>, bool>::iterator cachedPermissions = m_originAndPermissionsCache.find(permissionInfo);
	if (cachedPermissions != m_originAndPermissionsCache.end())
	{
		bool accepted = cachedPermissions->second;
		if (accepted)
		{
			callback->Continue(requested_permissions);
		}
		return accepted;
	}
```


https://github.com/ThinBridge/Chronos/pull/134/files#diff-8aad731b958e24ed51cb70eac250f7763e1f8110fadf4608f54c205a01a6845bR1903
```
	m_originAndPermissionsCache[permissionInfo] = true;
```

Reply:
Thanks, fixed it.